### PR TITLE
Add audio capture stats, rename method to getStats.

### LINKED ISSUE #96: Migrate capture metrics from RTCAudioSourceStats to MediaStreamTrack method
The [RTCAudioSourceStats](https://w3c.github.io/webrtc-stats/#audiosourcestats-dict*) contains capture related metrics that are better suited for a MediaStreamTrack method than RTCPeerConnection.getStats() since you may be capturing for a non-WebRTC context.

This was covered in the [April 18, 2023 Virtual Interim](https://docs.google.com/presentation/d/1pt1iEtqv49etzQ_u8E4QOpE3MUqQHbH6H8KlSW7MpfA/edit#slide=id.g22e9343079e_1_1) and the Working Group input was supportive: Moving them to the track is the right direction, and no strong opinions on the name of the method.

There was even preference to rename track.getFrameStats() to track.getStats() expressed and this would be a good opportunity to do so, before the video capture stats are implemented.

Original issue over on the WebRTC stats spec: https://github.com/w3c/webrtc-stats/issues/741

### PR DESCRIPTION
**Pull request by henbos**
Fixes #96.

***
<a href="https://pr-preview.s3.amazonaws.com/henbos/mediacapture-extensions/pull/97.html" title="Last updated on Apr 20, 2023, 2:55 PM UTC (c44f477)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-extensions/97/13addf4...henbos:c44f477.html" title="Last updated on Apr 20, 2023, 2:55 PM UTC (c44f477)">Diff</a>

### CODE REVIEW

**Comment by henbos** [comment]
This is approximately just a move, but instead of talking about "audio samples" I've rephrased it to match the terminology that is used outside of webrtc-stats to say "audio frames" instead. I've also added a clarifying note.

File: index.html [hunk 1]
```diff
@@ -419,56 +419,116 @@ <h3>Transferable MediaStreamTrack</h3>
...
+            a camera, total video frames is incremented by frames produced by the
             camera. If no frames are flowing, such as if the track is muted or
-            disabled, then total frames does not increment.</p>
+            disabled, then total video frames does not increment.</p>
+          </div>
+        </li>
+      </ul>
+      <p>If a {{MediaStreamTrack}} is sourced from <code>getUserMedia()</code>
+      and {{MediaStreamTrack/kind}} is <code>"audio"</code>, the user agent is
+      required to keep track of the following based on audio frames from its
```
Comment (jan-ivar) [inline index.html:483]:
```suggestion
      and {{MediaStreamTrack/kind}} is <code>"audio"</code>, the user agent
      MUST keep track of the following based on audio frames from its
```

**Review by jan-ivar — APPROVED** [review]
LGTM in the sense that this is a move from stats, which I appreciate! 

But I am not an audio expert, and would appreciate @padenot looking this over.

**Comment by jan-ivar** [comment]
`Editors' can integrate` pending approval from @padenot.

File: index.html [hunk 2]
```diff
@@ -419,56 +419,116 @@ <h3>Transferable MediaStreamTrack</h3>
...
+          <div class="note">
+            <p>The average capture delay per audio frame can be calculated as
+            [= total audio capture delay =] / [= total audio frames captured =].</p>
+          </div>
+          <div class="note">
+            <p>If the track is unmuted and enabled, the total audio frame duration and
+            total audio frames captured are increased by the audio frames
+            produced by the audio capture device. If no audio frames are
+            flowing, such as if the track is muted or disabled, then the audio
+            duration, captured frames and dropped frames does not increase.</p>
```
Comment (jan-ivar) [inline index.html:531]:
```suggestion
            duration, captured frames and dropped frames do not increase.</p>
```

**Review by youennf — APPROVED** [review]

**Comment by jan-ivar** [comment]
@dontcallmedom can we add @padenot as a reviewer in this repo? I don't seem to have the powers.

**Comment by jan-ivar** [comment]
I've filed #98 with Paul's concerns.

**Comment by henbos** [comment]
#98 is about async versus sync, not about the metrics added in this PR. Can we follow up on the API shape separately from the addition of these metrics? Can I merge the PR?

**Comment by henbos** [comment]
> https://github.com/w3c/mediacapture-extensions/issues/98 is about async versus sync, not about the metrics added in this PR. Can we follow up on the API shape separately from the addition of these metrics? Can I merge the PR?

Ping @jan-ivar

**Comment by henbos** [comment]
This PR is obsolete. I will close it once I have a synchronous version of the proposal and other PRs have landed. Iceboxed in the meantime

**Comment by henbos** [comment]
Closing this one, will upload a new version which follows the new API pattern and rephrases things a bit

**Pull request closed without merging**